### ISSUE TITLE
[Discussion] 10C/OEGD Why is the checkbox necessary for the patient to receive her test-result in the CWA? 

### ISSUE THREAD

**Issue by alanrick** [corona-warn-app/cwa-documentation#475]
## Your Question
Why does the test-center need to mark a checkbox in order for the patient to receive the test-result in the CWA? 

If the patient wants to receive the result in the CWA, for example to be informed sooner and without overloading the health center or telekom hotlines, then they should be able to make this decision themselves. Somebody else marking the checkbox does not add value (privacy etc) to the process and is major a source of error.  **[~50%](https://www.coronawarn.app/assets/documents/2020-12-04-cwa-daten-fakten.pdf) of positive results are not returned to the app, and this extraneous checkbox could be a major contributor to this poor result.**

According to the [CORONA-WARN-APP SOLUTION ARCHITECTURE](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md#upload-schedule-for-diagnosis-keys),  and in particular figure 2 in the documentation, the process for receiving the test result in the CWA consists of the following actions:
![figure_2](https://user-images.githubusercontent.com/7034358/101283790-1421ed80-37dd-11eb-8d95-824c42254b23.png)

A1. Step 1: User receives test-kit (S1-1) and scans the QR-code with the CWA app (S1-2) to prepare receiving the unique Test-ID.
A2. CWA stores the Test-ID locally on the mobile phone as part of the CWA App data and transfers it to a server  (S1-3) to subscribe to the result.
A3. User performs test
A4. User informs the test-center that they want the test-results on the CWA and later (or immediately) a member of the  test-center marks the 10C/OEGD checkbox.
A5. Step 2: Test and 10C/OEGD (including QR-code and patient details) is sent to laboratory
A6. Laboratory process test
A7. Step 3: Personal details are extracted from the 10C/OEGD and sent with the test result to the health center (if positive) and the test-center (always)
A8. Test-ID (extracted from QR-Code) and test result (positive/negative) is sent to the Laboratory Information System, but without any personal details. From there is is transferred to the Test Result Server, again without any patient details.
A9. Step 4a: CWA queries the Test Result Server for the test result referenced by Test-ID (see action A2)
A10. If query is successful, the result of the test is shown in that user's CWA.

My interpretation is that the Test Result server does not contain any information about the patient. It just contains the Test ID,  and the test result, and perhaps a timestamp.

So the test result in the Test Result Server is anonymous. Furthermore, it contains no data about the patient, only the Test-ID and Test result which cannot be used to identify a person. It is totally anonymous just as it it was totally anonymous before reaching the test-center. Only the patient's CWA (noone elses) can determine if it matches, and only then if the patient has previously scanned the QR-Code into the CWA. Before scanning the patient will have accepted the ToU (AGB) of the CWA and been informed in different ways (wizard, graphics...)  about all the consequences and is in a position to determine whether or not to scan and receive the results in the app.

So why does the 10C/OEGD form need the test-center to mark the checkbox in order for it to be sent to the Test Result Server anonymously?

Reading the text of the checkbox on the [10C](https://www.kbv.de/media/sp/Ansichtsexemplar_Muster10C.pdf) doesn't help.

"Das Einverständnis des Versicherten zum Übermitteln des Testergebnisses" für Zwecke der Corona Warn-App auf den vom RKI betriebenen Server wurde erteilt. Dem versicherten wurden Hinweise zum Datenschutz ausghändigt."

Why is permission for this server required and not other servers, which may be hosted on third-party Cloud platforms? Why is the test-center allowed to make this statement without a signature from the patient? Surely the privacy agreement is part of the app and up-to-date in the app, rather than relying on a copy (out-of-date) in the test-center....

**Bottom line - This checkbox prevents the test result being returned when it is accidentally/hurriedly left unchecked even when the patient wants the results in their CWA app; but what value does it add?**

**Comment by alanrick** [comment]
@dsarkar why is this a discussion item? 
The checkbox can’t have been added to the forms by accident. Someone must have made the decision and have an answer.

**Comment by heinezen** [comment]
Hey @alanrick ,

I've checked for more information on why the checkbox text is phrased like it is and why it is necessary, but the design of the test form is not something we (the CWA team) control. We can see if we get a more detailed explanation for the docs from the RKI. The reason the checkbox exists is most likely a privacy-related legal requirement.

However, I think there is a basic misunderstanding in your firs paragraph

> **~50% of positive results are not returned to the app[, and this extraneous checkbox could be a major contributor to this poor result.]**

If you are refering to the figure "Anzahl potenziell teilbarer Testergebnisse", then this is wrong. The 54% number is refering to the overall results **shared by users** after receiving a result returned to the app (i.e. 54% shared their personal random IDs). For the other 46%, the result was also returned to the app, but they did not decide to share it. If the checkbox is not ticked, the result is not uploaded to the test result server and would therefore not be included in this figure because the 54% is already the percentage of the total number of results reported to the app.

----
Corona-Warn-App Open Source Team

**Comment by ndegendogo** [comment]
> the figure "Anzahl potenziell teilbarer Testergebnisse", [...] The 54% number is refering to the overall results shared by users after receiving a result returned to the app (i.e. 54% shared their personal random IDs). For the other 46%, the result was also returned to the app, but they did not decide to share it.

@heinezen interesting. I was always wondering how to read this number.
So, users sharing their keys via TeleTAN are **not** covered by this number, right?

**Comment by alanrick** [comment]
Many thanks @heinezen for following up on this. Genuinely appreciated.

> If you are refering to the figure "Anzahl potenziell teilbarer Testergebnisse", then this is wrong. 

I was indeed. So I was wrong. Then in terms of the diagram you're saying is that:

54% = count of messages requesting TAN (message `(4)` in diagram step 4a) 
compared to 
count of messages returning a positive result  `(1)-return pos-success` in the same diagram.

I.e. the 46% (100-54) efficiency loss is through users who were alerted of the positive result in their app (after having scanned in the qr-code) but decided _not_ to share the results or _not_ noticing the positive result on their app.  This is what is specifically being addressed by the new reminder function in release 1.7.1 and tomorrow we can expect to see an improvement over the 56% success reported when it was released  three weeks ago.

What puzzles me is that if this is true, then according to Michael Boehm's dashboard and my back-of-the-envelope calculations there is no other loss of efficiency in the app process. Ie. all labs are now connected and all tests support the CWA qr-code, and the checkbox is always ticked correctly (despite comments elsewhere in this forum that some doctors _never_ tick it).  I'll recheck my calculations as I find this unlikely but I'll publish them here if anyone is interested.

(54% success 2 weeks ago. 56% 3 weeks ago)

**Comment by alanrick** [comment]
@ndegendogo the RKI description makes it clear that it includes the teletans so I adjusted my answer above accordingly.

Nevertheless I find this very difficult to believe because 
- someone making the effort to collect a teletan is unlikely to refuse publishing the result
- the statistic hasn't changed since the 1.7.1 reminder feature.  It was 54%-56% in the weeks following the release, which is identical to the weeks preceding this major update. 

@heinezen If, despite my doubts, the RKI is measuring exactly what you stated, would it be possible to ask the RKI for a measurement of the percent of results not returned to the app directly because of unticked checkboxes?  
I.e. proportion of result- replies (`message (1) response` in diagram step 4a) 
compared to
scanned qr-codes ('message (3) in diagram step 1)

Apparently 90% of labs are connected to the CWA servers so this ratio could be calibrated.
If you don't have time to do this fair enough, I'll try myself.

**Comment by ndegendogo** [comment]
> the RKI description makes it clear that it includes the teletans 

This is the confusing part.
54% of ... what, exactly?

QR process:
- test results are uploaded by the labs. Some are positive, (hopefully) most are negative. Each of them is related to a QR code.
- users scan their QR code to retrieve their results. Those with negative result are out of scope of this calculation.
- only a fraction of the remaining users (with positive result) share their keys, using the QR code.

So, 100% for QR process is number of downloaded test results / 54% is number of keys shared via QR code?

And then there are those who call the hotline and get a TeleTAN. No relationship to any test results on the server.
- those who did not get a QR code, because their lab is not yet integrated with cwa infrastructure
- those who decided for whatever reason not to use their QR code
- those who had technical problems to scan their QR code

What is 100% here? 100% is number of all requested TeleTANs / 54% is number of used TeleTANs?

@heinezen

**Comment by ndegendogo** [comment]
> would it be possible to ask the RKI for a measurement of the percent of results not returned to the app directly because of unticked checkboxes

@alanrick how can the RKI have knowledge of this number?

**Comment by alanrick** [comment]
The RKI publishes statistics every Friday so I assume they have access to the server metrics. I've tried to explain how this number can be calculated from the message flow measurements.

**Comment by ndegendogo** [comment]
> I've tried to explain how this number can be calculated from the message-flow monitoring.

I think what confuses me is: the number of users uploading their keys ("54%") is only loosely related to the number of downloaded positive test results ("100%"). With less QR codes and more TeleTANs used you could get above 100%.

> assume they have access to the server metrics

That's a valid assumption. After all, they are the 'owner' of the app.
However, the server never sees any test results for those where the box is not checked. These results are transmitted only the 'traditional' way to the doctor and to the health authority.

**Comment by Ein-Tim** [comment]
Let me quickly summarize, if I understand everything correctly, this question has to be answered, or?:
How many QR-Codes were registered in the App but the user never got a result?

**Comment by alanrick** [comment]
_Part a)_ was "Is there a legitimate reason for the checkbox?"
Yours is _part b)_, assuming there is no legitimate reason.
.....
So far I haven't seen _part a)_ answered. And speculation without justification that legal/privacy protection is behind it just gives data-privacy laws (which I strongly believe in) a bad reputation.

**Comment by heinezen** [comment]
> I think what confuses me is: the number of users uploading their keys ("54%") is only loosely related to the number of downloaded positive test results ("100%"). With less QR codes and more TeleTANs used you could get above 100%.

@ndegendogo I don't exactly understand what you mean here, but you cannot get above 100% with TeleTANs. The 100% is not the number of downloaded results, it is the number of positive results verified through the app (via QR code or TeleTAN as stated in the document). TeleTANs are verified individually by the verification server. Should have made this more clear when I said "reported" results.

> If, despite my doubts, the RKI is measuring exactly what you stated, would it be possible to ask the RKI for a measurement of the percent of results not returned to the app directly because of unticked checkboxes?

@alanrick I think we can fetch these numbers, but it would help if you stated what you want to find out with these numbers before I make the request-

> the statistic hasn't changed since the 1.7.1 reminder feature. It was 54%-56% in the weeks following the release, which is identical to the weeks preceding this major update.

Just to clarify: This percentage is skewed because it considers all `shared/total` results since 01.09.2020. The percentage of shared results per week was at ~50% the week before 1.7.1 released.

> "Is there a legitimate reason for the checkbox?"

I have checked the documentation and the answer is in the data privacy notice ([Link](https://www.coronawarn.app/assets/documents/cwa-privacy-notice-de.pdf)) in Question 3.

----
Corona-Warn-App Open Source Team

**Comment by ndegendogo** [comment]
@heinezen ... I think I start to understand now ...

In both process variants (QR code and TeleTAN), the user gets a **registration token**, and the number of all such registration tokens corresponds to 100% of shareable key sets.

In the **TeleTAN process**, the user has already decided to share their keys. So the fraction of missing uploads in this variant is *hopefully* very small, and we may attribute them all to technical problems or bad user interface of the app / or the process.

In the **QR process**, the fraction of missing uploads can have several reasons:
 * the user wants to get the lab result, but decides against sharing their keys
 * the user forgets sharing, or is not aware that they should do so, or does not know how 
 * user has technical problems to use the original QR code, but still shares their keys with TeleTAN
 * user has technical problems to use the original QR code, and therefore does not share their keys (either not knowing the TeleTan procedure, or gives up because the hotline is overloaded)

Strategy to improve fraction of users sharing their keys should address the specific reasons to be as effective as possible.

Is there any statistics available (or would it be possible to generate such a statistics) separately for each of the two procedures?

**Comment by ndegendogo** [comment]
> Just to clarify: This percentage is skewed because it considers all shared/total results since 01.09.2020.

Oh - a statistics of "current" / recent numbers could give much better insights ... is such a statistics available?

**Comment by alanrick** [comment]
> @alanrick I think we can fetch these numbers, but it would help if you stated what you want to find out with these numbers before I make the request

**Background: I want to know how much the efficiency of the CWA would increase if labs ignore this checkbox.**
This has been my burning question for the last few months. 

We know that:
a) there is no reason a patient would scan the test's qr-code if they didn't want the results in the app
b) the checkbox is not marked by the patient, but by the health staff, contrary to the privacy declaration
c) the checkbox is _always left unchecked_ by some test-centers and _always checked by others_ making it meaningless in many cases.
d) the test-result input into the CWA server is totally anonymous (hashed ID instead of the qr-code scanned)
e) if the patient changes their mind after scanning they can delete the test from their app before the result is returned
f) the test result remains anonymous in the CWA ecosystem (thanks to the hashing) until the instant the result is returned to the patient's app (if they haven't revoked permission), independent of whether the checkbox is set or not. At that point it is known only by the patient, no other part of the CWA eco-system
g) independent of the CWA eco-system, the test-result is anyway passed with the patient's data to the health authority, making a mockery of the CWA's stringent anonymous processing and the superfluous check-box.

> the answer is in the data privacy notice ([Link](https://www.coronawarn.app/assets/documents/cwa-privacy-notice-de.pdf)) in Question 3. 

I believe the answer is in section 12.b of this privacy notice. This makes it very clear that there is no need for an additional checkbox. In fact the checkbox is not mentioned anywhere in the whole privacy notice. 

**So my most pressing question is 
a) why, precisely, is the checkbox needed (and heeded)?
b) what percentage of results are failing to be returned directly to the patients due to the checkbox being incorrectly completed by health staff?**


> Just to clarify: This percentage is skewed because it considers all `shared/total` results since 01.09.2020. 

Yes, I'd realized that. And like @ndegendogo I was surprised that the RKI hadn't introduced a weekly chart so that the impact of the 1.7.1 release was not skewed. Nevertheless, I'd anticipated that in order for the RKI to suggest this new function the improvement would need to go up by at least 20% to make it worthwhile, and this should have shown up as a blip in the skewed results already.

Particularly with the new UK variant, the speed of returning the result to the patient is extremely important when fighting the spread of the disease. There are nearly a dozen EU Apps available, but **the German CWA App is the only EU app capable of returning the test result instantly to the user. This superfluous checkbox undermines the potential success of this feature.**

**Comment by codejus** [comment]
@alanrick 

>  I believe the answer is in section 12.b of this privacy notice. This makes it very clear that there is no need for an additional checkbox. In fact the checkbox is not mentioned anywhere in the whole privacy notice.

You and @heinezen are mixing up the CWA data protection and regulations for the labs and doctors.
You have to distinguish between:

1. The data processing of CWA and its infrastructure:
- In the onboarding you're giving consent in the data processing by the RKI (Legal controller acc. to Art. 4 Nr. 7 GDPR) as explained by the data protection statment @heinezen has linked to
- This consent does ONLY cover the processing inside the sphere of influence of the RKI and its contracted parties, which is in the end the CWA infrastruture and nothing more. Also the consent is only valid between you and the RKI --> This does not cover the Corona test in the laby/by your doctor as this is defnitly out of this sphere obviously.

2. The data processing by the labs and doctors

-  They are by definiton their own controller, because they are processing the personal/medical data for their own purposes like the potential needed treatment or billing with the insurances, etc.. So simplyfied they can't be just a "test service provider" for the RKI.
- So they need their own allowance (by law or consent) for the processing of personal data in general. There are many different ones in place here. (I don't want to get hier in depth.) But none of these does cover the sharing of the test result into the CWA Testresult server. In addition the test result is deemed as sensitive data in the sense or art. 9 I GDPR, so a consent is required for any sharing.
- In addition labs and doctor are subject to the doctor's patient confidentiality obligation (a breach is a criminal offence). This includes the lab result as medial information.
- -->The last both points mean that a consent for the sharing of the test result into the Testresult Server by the doctor/lab is needed. This consent is given by ticking field 9 of the 0C/OEGD forms.

To sum it up. The field 9 of the 0C/OEGD forms has nothing to do with the CWA itself and is thus completly out of scope for SAP/Telekom and even the RKI. Even if it looks different at first glance. This is also the reason @alanrick was note able to find anything about the 0C/OEGD forms in the CWA DPS and the mentioning of a DPS at the field 9.

The only way to get rid of field 9 is probably a law which regulates the CWA. But the government decided against this way.

Little disclaimer at the end, I have simplfied some things here and it is late.

Hope this brings some clarity regarding this issue.

**Comment by alanrick** [comment]
First off, many thanks for answering in so much detail and I won't dispute what you've described. 

I also understand that it is too complex for me to begin to understand in detail. 
So, skipping part a), @heinezen it would be terrific if the RKI could provide statistics of how much the checkbox deteriorates the effectivity of the app. This would at least be useful for the next pandemic.

>  a consent for the sharing of the test result into the Testresult Server by the doctor/lab is needed. This consent is given by ticking field 9 of the 0C/OEGD forms.

a) It does seem counterintuitive that this consent is not given by the patient. But I guess as you've explained in your answer the whole thing is too complex for the patient to grasp anyway.

b) It is also counterintuitive to me that consent is needed for providing data anonymously, but I've heard of scandals where anonymous data was fed into medical systems so there must be a good reason here.

c) Ideally, the test-lab could query the CWA server for the patient's approval, before submitting the result (my naïve assumption that the patients approval outweighs the health worker's) but I guess this is not possible due to the hashing of the qr-code (ironic).

d) It is counterintuitive to me that legal framework exists that cannot be enforced or even verified as being followed.

e) I wondered how the labs can report the results the health authorities but I guess that is covered by a separate legal framework (pandemic laws) and even though they may have the right to feed the test-result with qr-code back into the server to make its way to the app, that would involve all sorts of technology and process changes - but perhaps something to consider for the SORMAS system in future.


Hey, ho! It's a complex world we live in. My hopes that data-privacy concerns could be clearly formulated so the citizen can understand that they are in their best interests have been dashed. 

Many thanks @codejus for answering the question. And @heinezen many thanks for offering to ask about the statistics. If you get a chance, that would be great.

**Comment by heinezen** [comment]
> why, precisely, is the checkbox needed (and heeded)?

As I said, the answer is in Section 3 of the privacy notice. The CWA and all connected systems process health data under Art.9(2)(a) GDPR which requires explicit consent from the user/patient. A test result is such data, therefore we need consent to upload and store it on the test result server. The privacy notice emphasizes the required consent in section 6b)

>  For this to work, the testing laboratory needs to  be  connected  to  the  server  system  and,  as part  of  the testing  procedure, you must have given separate consent to your test result being sent.

It is not specified *how* exactly you give consent, but we have to ask for explicit consent for this specific purpose as required by the GDPR.

> what percentage of results are failing to be returned directly to the patients due to the checkbox being incorrectly completed by health staff?

I don't think there is a way to get numbers for this because there is no way to differentiate between a checkbox that is unticked because a patient has not given consent and a checkbox that was unticked because the staff made an error. The RKI can only investigate extreme irregular reporting behaviour.

> We know that: [...]

What you write here contains a lot of speculation, so I will carify:

* Doctors or test centers cannot decide on their own to mark the checkbox. If a test center always (or never) marks the checkbox, this is likely a breach of privacy. If you are aware of a test center doing this, please report it at datenschutz@rki.de so that it can be investigated.
* The test result is not anonymous, it is **pseudonymous** as explained in the form and in the privacy notice section 6b
* Passing the data to the local health authority is done under different legal requirements. The CWA and the local health authority are separate entities that process health data for different purposes. For the purposes of processing the test result in the CWA and on the test server, we require explicit consent.

**Comment by codejus** [comment]
> a) It does seem counterintuitive that this consent is not given by the patient. But I guess as you've explained in your answer the whole thing is too complex for the patient to grasp anyway.

As @heinezen inezen has already mentioned a pre checked form would be a data protection violation, because the processing of the data then happens without any allowance. Regarding the information part, every patient must be handed out the "Datenschutzerklärung", which is mentioned on the 0C/OEGD form. this DSE should normally be a different one than the CWA one and explains the sending of the test result into the CWA infrastructure. A valid DSE explains all the patient gives his consent to in clear manner, etc.

Ah a little sidenote, every time the lab hands someone a pre checked form and the test result gets transferred into CWA infrastructure they are required by law to report this date breach within 72h to the data protection authority. :D

> b) It is also counterintuitive to me that consent is needed for providing data anonymously, but I've heard of scandals where anonymous data was fed into medical systems so there must be a good reason here

If the test result would be anonymous, which is not the case because the Lab always knows which ID your QR code has, there would be needed no consent, as the GDPR and connected laws would then not be applicable, because no personal data is present. The most things normal people refer to as anonymous data are just pseudonymous data, because there is very often a way to establish a connection to a person, etc.

> e) I wondered how the labs can report the results the health authorities but I guess that is covered by a separate legal framework (pandemic laws) and even though they may have the right to feed the test-result with qr-code back into the server to make its way to the app, that would involve all sorts of technology and process changes - but perhaps something to consider for the SORMAS system in future.

The lab is required to report COVID-19 infections according to § 6 I 1 t) IfSG (The data protection legal allowance will be Art. 6 I c) GDPR in connection with § 6 I 1 t) IfSG) to the responsible health authority and happens independently from the CWA infrastructure via FAX or DEMIS (it the lab and the health authority is connected to it. You see it is the same problem as with CWA).

-----------
A little addition to my post from this night:

1. The doctor/lab needs you consent for the upload of the test result into the CWA Test result server, as the data is sent to another party.
2. RKI needs your consent to the connected data processing inside the CWA infrastructure in whole. Otherwise, a processing would be illegal.

So, it is not working without field 9. Probably there is an other way to work around that by app design etc. But the construction of such a solution will be not that easy and time consuming. Man legal and technical aspects would hav to be considered.

**Comment by alanrick** [comment]
@heinezen I think we agree on much of this but I'm struggling with the term pseudonymous which I couldn't find in the privacy agreement.
Step 1: Antje has the App, and Bernd and Carsten don't, and all three are tested (with Antje scanning the qr-code) 
Step 2: The doctor's assistant marks all checkboxes (by accident), sends the samples to the lab.
Step 3: The lab feeds all  test-results into the CWA server.
Is this test-result data on the CWA server pseudonymous?

My understanding was that  it only becomes identifiable once Antje connects to collect her results (like Schrodinger's cat), and even then on the server nothing has changed, the test-result is only identifiable on her own phone. And even then, nobody has anyway of determining from the data in the system which result belonged to Bernd, and which to Carsten.

So is pseudonymous = identifiable at a later stage, or through a software bug (e.g. Bernd's IP address accidentally preserved on the server) ?

**Comment by ndegendogo** [comment]
@alanrick well, everybody who knows the relationship between the true name and the QR code could attribute the test results to the person.

**Comment by alanrick** [comment]
@heinezen @codejus 
Just to add one point to what you have both written. My assumption is that the consent can be explicitly given in the app when scanning, but this doesn't help the lab who has no access to this. The lab cannot tell whether the test-sample came from someone using the app or not, and cannot see the consent in the app. Hence the checkbox on paper.

But I was hoping that this result could be treated as anonymous if it was stripped of all personal data when fed into the server.

That's why I was so hung up on the term pseudonymous.
Thanks for your support - and I guess you're right, @heinezen, no measurement from the RKI would clarify how much the checkbox is being left accidentally unchecked since theoretically someone using the app may scan the code but not want the results to go to the CWA ( even though I can't figure the logic behind such a behaviour.)

**Comment by alanrick** [comment]
> and the QR code

Yes. But not in the CWA ecosystem. And that was my only focus. Getting the test-result back to the patient as quickly as possible - the advantage of the CWA App over all other EU Corona apps.

**Comment by codejus** [comment]
> 
> 
> > and the QR code
> 
> Yes. But not in the CWA ecosystem. And that was my only focus. Getting the test-result back to the patient as quickly as possible - the advantage of the CWA App over all other EU Corona apps.

It doesn't matter who can make a connection between the QR code and the person behind it. So, a once pseudonymous data stays pseudonymous whether or not the processor can establish the connection to the natural person. This means they are still personal data and and the date protection regulations still apply.

Pseudonymisation is always only a securtiy measure to reduce the risk of the exposure to thind parties.

For a better understanding of the terms personal, pseudonymious and anonymised data I recommend this site here: [ICO- What is personal data?](https://ico.org.uk/for-organisations/guide-to-data-protection/guide-to-the-general-data-protection-regulation-gdpr/what-is-personal-data/what-is-personal-data/) Hope that helps to clarify.

**Comment by alanrick** [comment]
Thanks.  Thanks a good link.

**Comment by alanrick** [comment]
I'd like to close the question with an answer that is simple to understand but doesn't stonewall.

oooooooo

When a patient is tested, the result of the test can be returned to the patient's app. The result data is _personal data_ of especially sensitive nature so it requires strong protection. The result is pseudonymised to provide additional security and can be retrieved by noone other than the patient who scanned the test's qr-code during testing. Note: This data is pseudonymised, but not anonymous, so GDPR requirements still need to be fulfilled.

Although the patient's scanning action could be treated as explicit approval for storing the result temporarily in the CWA system (assuming the necessary accompanying text in the CWA screen and ToU/AGB), the lab entering the test-result in the system will NOT have access to this information, so a separate channel is needed to authorize the lab to enter the test result into the CWA system.  This is the 10C/OEGD field 9 checkbox, which the health-center is legally committed to complete according to the patient's will.

oooooooo

I realize I've missed much of the detail provided above (particularly the different legal frameworks) , and I suspect there are other aspects such as audit-trails which play a role, but is this a reasonable alternative to simply "legal requirements"?

**Comment by heinezen** [comment]
Since all questions in here seem to be answered, I'll close this issue.

----
Corona-Warn-App Open Source Team
